proposals/0000-cases.md

```diff
@@ -0,0 +1,641 @@
+      to lambda expressions but which could also have guards and multiple clauses,
+      it could be used instead of `-XLambdaCase` and would be able to match on
+      multiple patterns.
+ 2. The extension `-XMultiWayIf` essentially introduces standalone guards,
+    simplifying the use of guards that aren't at the outermost level of a
+    function declaration or case-expression. Among other things, this made it
+    easier to use guards inside of lambda expressions.
+    - If there were an expression similar to lambda expressions that could have
+      guards and wasn't required to have at
+      least one parameter, it could be used instead of `-XMultiWayIf`. This
+      includes all uses of `-XMultiWayIf`, not just those inside of lambdas
+      (see Example section).
+ 3. During the implementation of `-XLambdaCase`,
+    [some suggested](https://gitlab.haskell.org/ghc/ghc/issues/4359#note_51110)
+    allowing lambda expressions to have multiple clauses. This was not
+    implemented: The most obvious approach of turning `\` into a layout herald
+    had the disadvantage of making some common idioms invalid.
+    - This could be circumvented by introducing a new expression that isn't required
+      to be backwards compatible with existing idioms.
+
+## Proposed Change Specification
+
+Example:
+
+```haskell
+filter = \cases _ []                 -> []
+                p (x:xs) | p x       -> x : filter p xs
+                         | otherwise ->     filter p xs
+```
+
+A new extension `-XMultiWayLambda` is introduced. Under this extension, a new
+expression is enabled, introduced by the token sequence <tt>\\&nbsp;cases</tt>. The whitespace between  `\` and `cases`
+is optional and may contain an arbitrary sequence of whitespace characters.
+`\cases` behaves in a way largely similar to `\`, but it is a layout herald, can have multiple
+clauses, and may contain guards (see BNF for details).
+
+Zero clauses are not permitted, as the expression would be ambiguous.
+(See `Alternatives` section for details.)
+
+Like the existing behavior for alternatives in `case`- and
+`\case`-expressions, and equations in function declaration syntax, it is
+possible to use `where` clauses within each clause of a `\cases`-expression.
+
+As with function declaration equations, all clauses must have the same number of patterns.
+
+Given a `\cases`-expression `mcexp` with one or more scrutinees and a function
+`f` declared with function declaration syntax, and with the same alternatives
+and same guards for each alternative as `mcexp`, the semantics of the
+expression `mcexp` are the same as those of the expression `f`. If `mcexp` has
+no scrutinees, the semantics are the same as those of an expression `p`
+declared with a pattern binding with the same guards as `mcexp`.
+
+The new expression matches function declaration syntax very closely, making
+refactoring easy.
+
+#### BNF Changes
+
+**Bold** indicates changes to the existing BNF.
```

Comment:
Can you include the \case syntax too, to show off the similarities and differences?

Add some text, after the BNF saying

"As the BNF shows
* `\case` always has arity 1, whereas \cases can have any arity, including zero
* The patterns in `\case` do not need to be parenthesised (as in `case`), whereas the patterns in `\cases` must be atomic or parenthesised (as in function definitions)
* `\case` can have zero clauses, whereas `\cases` must have at least one clause."

You may think of other points to make